tests/integration/ranker_training/offline/yaml/trainer.py

```diff
@@ -0,0 +1,55 @@
+import math
+import pickle
+from pathlib import Path
+
+import numpy as np
+
+from jina.executors.rankers.trainer import RankerTrainer
+
+
+class RandomWeightedRankerTrainer(RankerTrainer):
+
+    WEIGHT_SCALE = 1 / max(1.0, (2 + 2) / 2.0)
+    WEIGHT_LIMIT = math.sqrt(3.0 * WEIGHT_SCALE)
+    WEIGHT_FILENAME = 'weights.txt'
+
+    def __init__(self, params: dict = None, weights_shape: float = (2, 2)):
+        super().__init__()
+        self._params = params
+        self._weights_shape = weights_shape
+        self._weights = np.random.uniform(
+            -self.WEIGHT_LIMIT, self.WEIGHT_LIMIT, size=weights_shape
+        )
+
+    def train(self, *args, **kwargs):
```

Comment:
it would be nice to see how the `trainer` actually receives data about matches and trains accordingly